.github/actions/release-container/action.yaml

```diff
@@ -0,0 +1,311 @@
+    required: false
+
+  cosign-password:
+    description: "Private key password for cosigning images."
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: clone repo
+      uses: actions/checkout@v3
+
+    # setup
+    - name: install cosign
+      if: inputs.cosign-public-key != ''
+      uses: sigstore/cosign-installer@v3.1.2
+
+    - name: forward proxy settings
+      shell: bash
+      run: |
+        echo "HTTP_PROXY=$HTTP_PROXY" >> $GITHUB_ENV
+        echo "HTTPS_PROXY=$HTTPS_PROXY" >> $GITHUB_ENV
+        echo "NO_PROXY=$NO_PROXY" >> $GITHUB_ENV
+
+    - name: install buildx
+      uses: docker/setup-buildx-action@v3
+      with:
+        driver-opts: |
+          env.http_proxy=${{ env.HTTP_PROXY }}
+          env.https_proxy=${{ env.HTTPS_PROXY }}
+          "env.no_proxy='${{ env.NO_PROXY }}'"
+
+    - name: login to container registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.registry-user }}
+        password: ${{ inputs.registry-token }}
+# remainder
+    - name: determine release
+      uses: docker://ghcr.io/codfish/semantic-release-action:v3
+      id: semantic
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      with:
+        dry-run: true
+        branches: ${{ inputs.release-branches }}
+        tag-format: ${{ inputs.release-tag-format }}
+        plugins: |-
+          [
+            ['@semantic-release/commit-analyzer', {
+              "releaseRules": ${{ inputs.release-rules }},
+            }], 
+            '@semantic-release/release-notes-generator',
+            ['@semantic-release/github', {
+              "successComment": false,
+              "failTitle": false
+            }]
+          ]
+
+    - name: annotate the ci run with release version output
+      if: steps.semantic.outputs.new-release-published == 'true'
+      shell: bash
+      run: >-
+        echo "::notice title=Job: ${{ github.job }} / ${{ inputs.job-name }}::%0A
+        Next release will be version ${{ steps.semantic.outputs.release-version }}"
+
+    - name: release container metadata
+      if: steps.semantic.outputs.new-release-published == 'true'
+      uses: docker/metadata-action@v5
+      id: meta
+      with:
+        images: |
+          ${{ inputs.registry }}/${{ inputs.registry-repo }}
+        tags: |
+          type=raw,value=${{ inputs.registry-tag-prefix }}${{ steps.semantic.outputs.release-version }}${{ inputs.registry-tag-suffix }}
+
+    - name: build and push
+      uses: docker/build-push-action@v5
+      id: build
+      with:
+        file: ${{ inputs.build-file }}
+        context: ${{ inputs.build-context }}
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        push: true
+        github-token: ${{ github.token }}
+        cache-from: type=gha
+        cache-to: type=gha,mode=max
+        provenance: true
```

Comment:
Add `load: true` so that the build image is directly available to the test task that might run after.